src/inspector/node_string.cc

```diff
@@ -62,9 +71,14 @@ String StringViewToUtf8(v8_inspector::StringView view) {
   size_t expected_utf8_length =
       simdutf::utf8_length_from_utf16(source, view.length());
   MaybeStackBuffer<char> buffer(expected_utf8_length);
+  // convert_utf16_to_utf8 returns zero in case of error.
   size_t utf8_length =
       simdutf::convert_utf16_to_utf8(source, view.length(), buffer.out());
-  CHECK_EQ(expected_utf8_length, utf8_length);
+  // We have that utf8_length == expected_utf8_length if and only
+  // if the input was a valid UTF-16 string. Otherwise, utf8_length
+  // must be zero.
+  assert(utf8_length == 0 || utf8_length == expected_utf8_length);
```

Comment:
If we are consistent with the current implementations:

```suggestion
  CHECK(utf8_length == 0 || utf8_length == expected_utf8_length);
```